src/methods/to.js

```diff
@@ -428,7 +428,16 @@ export class ToImplementation {
         throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError);
 
       default:
-        obj = arg.throwIfNotConcreteObject();
+        if (realm.isInPureScope()) {
+          // Create a placeholder value to represent the ObjectValue that we would've
+          // received, but this object should never leak so as an optimization we will
+          // let operations on top of this object force the ToObject operations instead.
+          obj = AbstractValue.createFromType(realm, ObjectValue, "sentinel ToObject");
```

Comment:
I still think that this should be createFromBuildFunction and that the serializer should just work without needing another special case. I'm willing to let this go for now and fix it up in a subsequent pull request.

Reply:
To be able to do that safely, and also stay performant/small, we would need to be able to determine that this did in fact never leak and that this value is only referenced once. That way we can remove the explicit ToObject operation as an optimization in the serializer. That seems like a lot of work for something that currently can't happen but might be something we need for other reasons and then we can change this back.